###################################################################################################
Date: Tue, 01 July 2025

Subject: Observing bluetoothd crash in setconf_cb function

NULL checks added to avoid dereferrencing NULL pointers

Source: Bluez open source
Upstream-Status: Pending
Signed-off-by:Darshan Desale Darshan_Desale@comcast.com 
###################################################################################################
Index: bluez-5.48/profiles/audio/avdtp.c
===================================================================
--- bluez-5.48.orig/profiles/audio/avdtp.c
+++ bluez-5.48/profiles/audio/avdtp.c
@@ -1424,6 +1424,11 @@ static void setconf_cb(struct avdtp *ses
 {
 	struct conf_rej rej;
 	struct avdtp_local_sep *sep;
+	if(session == NULL){;
+	        error("session is NULL"); // Handle session if NULL
+	        DBG("session is NUll");
+	        return;
+	}
 
 	if (err != NULL) {
 		rej.error = AVDTP_UNSUPPORTED_CONFIGURATION;
